lib/compat/wordpress-6.1/hook-theme-json-for-classic-themes.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+function theme_json_current_theme_has_support( ){
+	return WP_Theme_JSON_Gutenberg::has_theme_support();
+}
+
+function theme_json_default_filter_for_classic_themes( $theme_json_data ) {
+	if ( theme_json_current_theme_has_support() ) {
+		$new_data = array(
+			'version'  => 2,
+			'settings' => array( /* we should maintain the presets by core */ ),
+			'styles'   => array( /* clear this? add only the base-layout-styles? */ ),
```

Comment:
As a bit of background, the subtlety behind `base-layout-styles` is that it only outputs layout styles and skips calling `get_block_classes`. The logic is around here: https://github.com/WordPress/gutenberg/blob/trunk/lib/compat/wordpress-6.1/class-wp-theme-json-6-1.php#L679-L681

So it's slightly different to filtering the `styles` object.

That said, the original reason for `base-layout-styles` was to ensure that we can output base / root layout styles for things like the gap between Buttons, without pulling in / outputting styles from the styles object. Given that #43981 is proposing outputting styles for classic themes, too, then the concept of `base-layout-styles` might become redundant anyway 🙂